src/tools/quemap/qzip.c

```diff
@@ -161,91 +161,117 @@ static void AddMaterials(const char *path) {
 
 	AddAsset(path, true); // add the materials file itself
 
-	buf = buffer;
+	parser_t parser;
+	Parse_Init(&parser, buffer, PARSER_DEFAULT);
 
 	num_frames = 0;
 	memset(texture, 0, sizeof(texture));
 
 	while (true) {
 
-		c = ParseToken(&buf);
-
-		if (*c == '\0') {
+		if (!Parse_Token(&parser, PARSE_DEFAULT, token, sizeof(token))) {
 			break;
 		}
 
 		// texture references should all be added
-		if (!g_strcmp0(c, "texture") || !g_strcmp0(c, "diffuse")) {
-			c = ParseToken(&buf);
-			if (*c == '#') {
-				g_strlcpy(texture, ++c, sizeof(texture));
+		if (!g_strcmp0(token, "texture") || !g_strcmp0(token, "diffuse")) {
+
+			if (!Parse_Token(&parser, PARSE_DEFAULT | PARSE_NO_WRAP, token, sizeof(token))) {
+				return;
+			}
+
+			if (*token == '#') {
+				g_strlcpy(texture, token + 1, sizeof(texture));
```

Comment:
Heh, I have another ticket pending to completely rewrite this to use Cm_LoadMaterials.. but this is fine.